internal/app/init/main.go

```diff
@@ -287,54 +307,60 @@ func startKubernetesServices(data *userdata.UserData) {
 	)
 }
 
+func reboot() {
+	// See http://man7.org/linux/man-pages/man2/reboot.2.html.
+	unix.Sync()
+	// nolint: errcheck
+	unix.Reboot(rebootFlag)
+
+	if *inContainer {
+		return
+	}
+
+	select {}
```

Comment:
Why the empty select here? Just to 'halt' all execution for the reboot?

Reply:
yep, it was present before my changes in `func main()`, I moved it here